mocpy/moc.py

```diff
@@ -2,427 +2,360 @@
+        nside : int, optional
+            256 by default
+
+        Returns
+        -------
+        result : `~mocpy.moc.MOC`
+            the created moc
+
         """
-        self._counter += 1
-        # force consistency to prevent too large interval array
-        if self._counter == 1000:
-            self._ensure_consistency()
-            self._counter = 0
 
-        if order > MOC.HPY_MAX_NORDER:
-            raise Exception('norder can not be greater than MOC max norder')
-        
-        if not nest:
-            ipix = ring2nest(1 << order, ipix)
+        nside_possible_values = (8, 16, 32, 64, 128, 256, 512)
+        if nside not in nside_possible_values:
+            raise ValueError('Bad value for nside. Must be in {0}'.format(nside_possible_values))
 
-        p1 = ipix
-        p2 = ipix + 1
-        shift = 2 * (MOC.HPY_MAX_NORDER - order)
+        result = cls.from_ivorn('ivo://CDS/' + table_id, nside)
+        return result
 
-        self._interval_set.add( ( p1 << shift, p2 << shift) )
+    MOC_SERVER_ROOT_URL = 'http://alasky.unistra.fr/MocServer/query'
 
-    def to_uniq_interval_set(self):
-        """
-        Return an IntervalSet in NUNIQ norder
-        reflecting the current state of the MOC
+    @classmethod
+    def from_ivorn(cls, ivorn, nside=256):
         """
-        r2 = IntervalSet(self._interval_set)
-        r3 = IntervalSet()
-        res = IntervalSet()
+        Create a `~mocpy.moc.MOC` object from a given ivorn
 
-        intervals = self._interval_set.intervals
-        if len(intervals)==0:
-            return res
-        
-        max_res_order = MOC.HPY_MAX_NORDER
+        Parameters
+        ----------
+        ivorn : str
+        nside : int, optional
+            256 by default
 
-        for order in xrange(0, max_res_order):
-            if r2.empty():
-                return res
-            
-            shift = long(2 * (max_res_order - order))
-            ofs  = (long(1)<<shift)-1
-            ofs2 = long(1)<<(2*order+2)
-            
-            r3.clear()
-            for iv in r2.intervals:
-                a = (long(iv[0])+ofs)>>shift
-                b= iv[1]>>shift
-                r3.add((a<<shift, b<<shift))
-                res.add((a+ofs2,b+ofs2))
-            
-            if not r3.empty():
-                r2 = r2.difference(r3)
-      
-        return res
+        Returns
+        -------
+        result : `~mocpy.moc.MOC`
+            the created moc
 
-    def add_position(self, ra, dec, max_norder):
-        """
-        add the HEALPix bin containing the (ra, dec) position
         """
-        theta, phi = utils.radec2thetaphi(ra, dec)
-        ipix = ang2pix(2**max_norder, theta, phi, nest=True)
 
-        if isinstance(ipix, np.ndarray):
-            for i in range(ipix.shape[0]):
-                self.add_pix(max_norder, ipix[i])
-        else:
-            self.add_pix(max_norder, ipix)
+        try:
+            from urllib import urlencode
```

Comment:
Move such code to the py 2 / 3 compat module and keep the rest of the code clean py 3?